tools/mc-java/src/test/java/io/spine/tools/mc/java/gradle/AnnotationProcessorConfigurationTest.java

```diff
@@ -26,55 +26,46 @@
 
 package io.spine.tools.mc.java.gradle;
 
-import com.google.common.testing.NullPointerTester;
 import io.spine.testing.UtilityClassTest;
 import io.spine.tools.mc.java.gradle.given.StubProject;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.ConfigurationContainer;
-import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 import static com.google.common.truth.Truth.assertThat;
 import static io.spine.tools.gradle.ConfigurationName.annotationProcessor;
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNull;
 
-@DisplayName("`PreprocessorConfig` should")
+@DisplayName("`AnnotationProcessorConfiguration` should")
 class AnnotationProcessorConfigurationTest
         extends UtilityClassTest<AnnotationProcessorConfiguration> {
 
-    private Project project;
-    private ConfigurationContainer projectConfigs;
-    private Configuration preprocessorConfig;
+    private static Project project = null;
+    private static ConfigurationContainer configurations = null;
+    private static Configuration preprocessorConfig = null;
 
     AnnotationProcessorConfigurationTest() {
         super(AnnotationProcessorConfiguration.class);
     }
 
-    @BeforeEach
-    void setUp() {
-        project = StubProject.createFor(getClass()).get();
-        projectConfigs = project.getConfigurations();
-        preprocessorConfig = projectConfigs.getByName(annotationProcessor.value());
+    @BeforeAll
+    static void createProject() {
+        project = StubProject.createFor(AnnotationProcessorConfigurationTest.class).get();
+        configurations = project.getConfigurations();
+        preprocessorConfig = configurations.getByName(annotationProcessor.value());
     }
```

Comment:
Does this approach lead to any side effects? As far as I understand, the project is now created only once and a series of tests are run on the same project. Even though it improves the performance, I'd probably double-check the code to be side-effect-free, or maybe add some notes on similar usages.

Reply:
I understand your concerns. There were mine too, and I've stumbled upon a couple of such side effects. I handled them, and I'm going to continue in another PR.

I don't see any other ways for improving the build performance when working with Gradle things.